cmd/tailscale/cli/serve_dev.go

```diff
@@ -266,6 +266,9 @@ func (e *serveEnv) runServeCombined(subcmd serveMode) execFunc {
 		if turnOff {
 			err = e.unsetServe(sc, dnsName, srvType, srvPort, mount)
 		} else {
+			if err := validateConfig(sc, srvPort, srvType); err != nil {
```

Comment:
We should move this check to immediately after we fetch the serve config to avoid mutating between fetching and doing this check, which I believe is happening [here](https://github.com/tailscale/tailscale/pull/9396/files#diff-d8fbd814bb4009e6070207323c70fcd9840a03f7a173d0aea0cec79ff943506dR261).